rust/rsc/src/rsc/blob.rs

```diff
@@ -1,71 +1,96 @@
 use crate::config::RSCConfig;
 use crate::types::GetUploadUrlResponse;
-use axum::{extract::Multipart, Json};
+use axum::{body::Bytes, extract::Multipart, http::StatusCode, BoxError, Json};
 use sea_orm::DatabaseConnection;
 use std::sync::Arc;
 use tracing;
 
-#[tracing::instrument]
-pub async fn get_upload_url(
-    _conn: Arc<DatabaseConnection>,
-    config: Arc<RSCConfig>,
-) -> Json<GetUploadUrlResponse> {
-    let url = config.server_addr.clone() + "/blob";
-    Json(GetUploadUrlResponse { url })
-}
+use futures::{Stream, TryStreamExt};
+use tokio::fs::File;
+use tokio::io::BufWriter;
+use tokio_util::io::StreamReader;
 
-pub async fn create_blob(mut multipart: Multipart, config: Arc<RSCConfig>) -> hyper::StatusCode {
-    let mut hash: Option<String> = None;
-    let mut contents: Option<axum::body::Bytes> = None;
+// to prevent directory traversal attacks we ensure the path consists of exactly one normal
+// component
```

Comment:
Can you explain this attack to me?

edit: Ah, I see you currently allow the user to name the file. That should not be a part of the protocol at all, the file should be named by the hash and a temporary should be used.